url/resources/a-element-origin.js

```diff
@@ -1,14 +1,4 @@
-var setup = async_test("Loading data…")
-setup.step(function() {
-  var request = new XMLHttpRequest()
-  request.open("GET", "urltestdata.json")
-  request.send()
-  request.responseType = "json"
-  request.onload = setup.step_func(function() {
-    runURLTests(request.response)
-    setup.done()
-  })
-})
+promise_test(() => fetch("resources/urltestdata.json").then(res => res.json()).then(runURLTests), "Loading data…");
```

Comment:
This kind of thing suggests that maybe we could make use of async setup, something like `setup(async () => ....)` which if it fails counts as a harness error. Would you use that?

Reply:
Yeah, that seems useful. I think this pattern probably warrants an abstraction too. `setup_from_json` or some such, but I haven't really been bothered enough to make it happen.